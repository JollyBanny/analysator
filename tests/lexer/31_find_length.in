var x1,y1,x2,y2,d:double;
begin
     writeln('Enter A(X1,Y1) Ð¸ B(X2,Y2)');
     readln( x1,y1,x2,y2 );
     d:=sqrt(sqr(y2-y1)+sqr(x2-x1));
     writeln('length segment |AB|=',d);
end.