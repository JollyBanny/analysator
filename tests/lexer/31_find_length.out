1	1	Keyword	Var	var
1	5	Identifier	x1	x1
1	7	Separator	Comma	,
1	8	Identifier	y1	y1
1	10	Separator	Comma	,
1	11	Identifier	x2	x2
1	13	Separator	Comma	,
1	14	Identifier	y2	y2
1	16	Separator	Comma	,
1	17	Identifier	d	d
1	18	Separator	Colon	:
1	19	Identifier	double	double
1	25	Separator	Semicolom	;
2	1	Keyword	Begin	begin
3	6	Identifier	writeln	writeln
3	13	Separator	Lparen	(
3	14	String	Enter A(X1,Y1) и B(X2,Y2)	'Enter A(X1,Y1) и B(X2,Y2)'
3	41	Separator	Rparen	)
3	42	Separator	Semicolom	;
4	6	Identifier	readln	readln
4	12	Separator	Lparen	(
4	14	Identifier	x1	x1
4	16	Separator	Comma	,
4	17	Identifier	y1	y1
4	19	Separator	Comma	,
4	20	Identifier	x2	x2
4	22	Separator	Comma	,
4	23	Identifier	y2	y2
4	26	Separator	Rparen	)
4	27	Separator	Semicolom	;
5	6	Identifier	d	d
5	7	Operator	Assign	:=
5	9	Identifier	sqrt	sqrt
5	13	Separator	Lparen	(
5	14	Identifier	sqr	sqr
5	17	Separator	Lparen	(
5	18	Identifier	y2	y2
5	20	Operator	Sub	-
5	21	Identifier	y1	y1
5	23	Separator	Rparen	)
5	24	Operator	Add	+
5	25	Identifier	sqr	sqr
5	28	Separator	Lparen	(
5	29	Identifier	x2	x2
5	31	Operator	Sub	-
5	32	Identifier	x1	x1
5	34	Separator	Rparen	)
5	35	Separator	Rparen	)
5	36	Separator	Semicolom	;
6	6	Identifier	writeln	writeln
6	13	Separator	Lparen	(
6	14	String	length segment |AB|=	'length segment |AB|='
6	36	Separator	Comma	,
6	37	Identifier	d	d
6	38	Separator	Rparen	)
6	39	Separator	Semicolom	;
7	1	Keyword	End	end
7	4	Separator	Dot	.
7	5	EOF
